src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/AddressNotAvailableException.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace RedHat.AspNetCore.Server.Kestrel.Transport.Linux
+{
+    public class AddressNotAvailableException : Exception
```

Comment:
Use this one instead: https://github.com/aspnet/AspNetCore/blob/0da5fabdbdd462cdd9528398efb6b1ffb9b27f65/src/Servers/Connections.Abstractions/src/Exceptions/AddressInUseException.cs#L8

Reply:
If I use ```AddressInUseException```

ASP.NET Core will rethrow with ```IOException```

See:
https://github.com/aspnet/AspNetCore/blob/4ef204e13b88c0734e0e94a1cc4c0ef05f40849e/src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs#L101-L113

Reply:
Should I do this?
```suggestion
    public class AddressNotAvailableException : InvalidOperationException
```

Reply:
> If I use AddressInUseException
ASP.NET Core will rethrow as IOException
See:
https://github.com/aspnet/AspNetCore/blob/4ef204e13b88c0734e0e94a1cc4c0ef05f40849e/src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs#L101-L113

Have you tried? I think that code path isn't used here, because the AspNetCore Libuv Transport is throwing AddressInUseException.

Reply:
I try with ASP.NET Core 2.1 and if I use ```AddressInUseException```  I  continue with error, I didn't try with ASP.NET Core 3.0 because I only can reproduce it in docker and .NET Core 3.0 + Docker + Linux 5.0.0-29 has a bug :(

Reply:
In Libuv, they throw when the error  is ```LibuvConstants.EADDRINUSE``` in ours case is ```PosixResult.EADDRINUSE``` already? Or ```PosixResult.EADDRNOTAVAIL`` minding same thing?

Reply:
Ah, yes. They are different.